rockcraft.yaml

```diff
@@ -8,20 +8,25 @@ base: ubuntu:22.04
 platforms:
   amd64:
 
-entrypoint: ["/bin/bash", "/opt/pebble/pebble-start.sh"]
-
-env:
-  - SPARK_HOME: /opt/spark
-  - SPARK_CONFS: /etc/spark8t/conf
-  - JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
-  - PYTHONPATH: /opt/spark/python:/opt/spark8t/python/dist:/usr/lib/python3/dist-packages
-  - PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/spark:/opt/spark/bin:/opt/spark/python/bin:/opt/spark-client/python/bin
-  - HOME: /home/spark
-  - KUBECONFIG: /home/spark/.kube/config
-  - SPARK_USER_DATA: /home/spark
+services:
+  spark-entrypoint:
+    command: "/bin/bash /opt/pebble/pebble-start.sh"
+    summary: "This is the service to startup Spark processes using the Spark entrypoint"
+    override: replace
+    startup: enabled
+    environment:
+      SPARK_HOME: /opt/spark
+      SPARK_CONFS: /etc/spark8t/conf
+      JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      PYTHONPATH: /opt/spark/python:/opt/spark8t/python/dist:/usr/lib/python3/dist-packages
+      PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/spark:/opt/spark/bin:/opt/spark/python/bin:/opt/spark-client/python/bin
+      HOME: /home/spark
+      KUBECONFIG: /home/spark/.kube/config
+      SPARK_USER_DATA: /home/spark
+    on-success: shutdown
+    on-failure: shutdown
```

Comment:
considering my comment in https://github.com/canonical/charmed-spark-rock/pull/23/files#r1229032686, if you add `user` and `group` here, then your entrypoint.sh could possibly be reduced to a single line